src/kudu/tserver/tablet_server_options.cc

```diff
@@ -29,32 +29,45 @@
 #endif
 #include "kudu/server/rpc_server.h"
 #include "kudu/tserver/tablet_server.h"
-#ifdef FB_DO_NOT_REMOVE
 #include "kudu/util/flag_tags.h"
 #include "kudu/util/status.h"
 
-DEFINE_string(tserver_master_addrs, "127.0.0.1:7051",
-              "Comma separated addresses of the masters which the "
-              "tablet server should connect to. The masters do not "
-              "read this flag -- configure the masters separately "
-              "using 'rpc_bind_addresses'.");
-TAG_FLAG(tserver_master_addrs, stable);
-#endif
+DEFINE_string(tserver_addresses, "",
+              "Comma-separated list of the RPC addresses belonging to all "
+              "instances in this cluster. "
+              "NOTE: if not specified, configures a non-replicated Master.");
+TAG_FLAG(tserver_addresses, stable);
 
 namespace kudu {
 namespace tserver {
 
 TabletServerOptions::TabletServerOptions() {
   rpc_opts.default_port = TabletServer::kDefaultPort;
 
-#ifdef FB_DO_NOT_REMOVE
-  Status s = HostPort::ParseStrings(FLAGS_tserver_master_addrs,
-                                    master::Master::kDefaultPort,
-                                    &master_addresses);
-  if (!s.ok()) {
-    LOG(FATAL) << "Couldn't parse tablet_server_master_addrs flag: " << s.ToString();
+  if (!FLAGS_tserver_addresses.empty()) {
+    Status s = HostPort::ParseStrings(FLAGS_tserver_addresses, TabletServer::kDefaultPort,
+                                      &tserver_addresses);
+    if (!s.ok()) {
+      LOG(FATAL) << "Couldn't parse the tserver_addresses flag('" << FLAGS_tserver_addresses << "'): "
+                 << s.ToString();
+    }
+    if (tserver_addresses.size() < 2) {
+      LOG(FATAL) << "At least 2 tservers are required for a distributed config, but "
+          "tserver_addresses flag ('" << FLAGS_tserver_addresses << "') only specifies "
+                 << tserver_addresses.size() << " tservers.";
+    }
+
+    // TODO(wdberkeley): Un-actionable warning. Link to docs, once they exist.
+    if (tserver_addresses.size() == 2) {
+      LOG(WARNING) << "Only 2 tservers are specified by tserver_addresses_flag ('" <<
+          FLAGS_tserver_addresses << "'), but minimum of 3 are required to tolerate failures"
+          " of any one tserver. It is recommended to use at least 3 tservers.";
+    }
   }
-#endif
+}
+
+bool TabletServerOptions::IsDistributed() const {
```

Comment:
Would it matter if more servers join the raft ring a little later via a config change and this function would give us stale information?

nit: `isStandaloneNode` might be a more indicative name for the function.

Reply:
Distributed is the opposite of standalone. This is what kudu also uses.
There is an assertion which checks that min # of servers is 2

Reply:
This option is only used in the very beginning of raft initialization to create the initial  fs and config, or validate the last config.